setup/StackSetup/PowerShellSetup.Test/MsiLayoutTests.cs

```diff
@@ -0,0 +1,274 @@
+
+namespace PowerShellSetup.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Management.Automation;
+    using System.Threading.Tasks;
+    using Xunit;
+    using Xunit.Abstractions;
+
+    /// <summary>
+    /// Set of tests to run against MSI
+    /// These set of tests are especially for scenario like:
+    ///     1) Layout errors, things getting copied to the wrong location
+    ///     2) Checking if files in the MSI including the MSI are signed
+    ///     3) Checking if files included in MSI uses right signing alogirthm
+    /// 
+    /// </summary>
+    public class MsiLayoutTests: MsiTestBase
+    {
+        const string MSI_NAME = "AzurePowerShell.msi";
+        const string MSI_EXTRACT_DIR_NAME = "msiContents";
+        string _procOutput;
+        string _procErr;
+
+        public MsiLayoutTests(ITestOutputHelper testOutput) : base(testOutput)
+        {
+            _procOutput = string.Empty;
+            _procErr = string.Empty;
+        }
+
+        [Fact]
+        [Trait("AcceptanceType", "CheckIn")]
+        public void VerifyMsiExecExists()
+        {
+            ProcessStartInfo psi = GetInitializedPSI();
+            psi.FileName = "Msiexec.exe";
+            psi.Arguments = "/qn /x:1234";
+
+            this.ExecuteShellCmd(psi, out _procOutput, out _procErr);
+            Assert.NotEmpty(_procOutput);
+        }
+        
+        //[Fact]
+        //[Trait("AcceptanceType", "CheckIn")]
+        //public void VerifyNoJavaScriptFiles()
+        //{
+        //    string procErr = string.Empty;
+        //    string msiContentsDirPath = ExtractMsiContents(out procErr);
+        //    IEnumerable<string> msiFiles = Directory.EnumerateFiles(msiContentsDirPath, "*.js", SearchOption.AllDirectories);
+        //    TestLog.WriteLine("Expecting no *.js files in MSI");            
+        //    foreach (string unsigFile in msiFiles)
+        //    {
+        //        TestLog.WriteLine(unsigFile);
+        //    }
+        //    Assert.Equal(0, msiFiles.Count<string>());
+        //}
+
+        //[Fact]
+        //[Trait("AcceptanceType", "CheckIn")]
+        //public void VerifyNoJsonFiles()
+        //{
+        //    string procErr = string.Empty;
+        //    string msiContentsDirPath = ExtractMsiContents(out procErr);
+        //    IEnumerable<string> msiFiles = Directory.EnumerateFiles(msiContentsDirPath, "*.json", SearchOption.AllDirectories);
+        //    TestLog.WriteLine("Expecting no *.json files in MSI");
+        //    foreach (string unsigFile in msiFiles)
+        //    {
+        //        TestLog.WriteLine(unsigFile);
+        //    }
+        //    Assert.Equal(0, msiFiles.Count<string>());
+        //}
+        
+        [Fact]
+        [Trait("SignedBuild", "BVT")]
+        public void VerifyFilesAreSigned()
```

Comment:
This automatically occurs as part of the signing task in msbuild,   Please remove